interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -817,6 +818,29 @@ public void computeEngineCreds(String serviceAccount, String oauthScope) throws
     assertEquals(goldenResponse, response);
   }
 
+  /** Test JWT-based auth. */
+  public void jwtTokenCreds(InputStream serviceAccountJson) throws Exception {
```

Comment:
I'm doing the "The client can optionally check the username matches the email address in the key file" which is more strict than the first part.